generators/server/templates/_Jenkinsfile

```diff
@@ -65,12 +65,16 @@ node {
     stage('clean') {
         sh "./gradlew clean"
     }
-
     <%_ if (applicationType !== 'microservice') { _%>
+        <%_ if (clientPackageManager === 'yarn') { _%>
+    stage('yarn install') {
+        sh "./gradlew yarn_install -PnodeInstall"
```

Comment:
I tested this with following configuration:
* Jenkins 2.45
* Docker 1.12.6
* Centos 7.3

It throwed error:
```
[4/4] Building fresh packages...
error /var/jenkins_home/workspace/jhipster-jenkins-test/node_modules/yo: Command failed.
Exit code: 1
Command: sh
Arguments: -c yodoctor
Directory: /var/jenkins_home/workspace/jhipster-jenkins-test/node_modules/yo
Output:
Yeoman Doctor
Running sanity checks on your system

✔ Global configuration file is valid
✔ NODE_PATH matches the npm root
✔ Node.js version
✔ No .bowerrc file in home directory
✔ No .yo-rc.json file in home directory
/var/jenkins_home/workspace/jhipster-jenkins-test/node_modules/semver-truncate/index.js:12
		throw new Error('Version ' + version + ' is not valid semver');
		^

Error: Version null is not valid semver
    at module.exports (/var/jenkins_home/workspace/jhipster-jenkins-test/node_modules/semver-truncate/index.js:12:9)
    at /var/jenkins_home/workspace/jhipster-jenkins-test/node_modules/bin-version-check/index.js:20:25
    at /var/jenkins_home/workspace/jhipster-jenkins-test/node_modules/bin-version/index.js:15:3
    at ChildProcess.exithandler (child_process.js:197:7)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
:yarn_install FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':yarn_install'.
> Process 'command '/var/jenkins_home/workspace/jhipster-jenkins-test/.gradle/nodejs/node-v6.9.5-linux-x64/bin/node'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

My .yo-rc.json:
```
{
  "generator-jhipster": {
    "jhipsterVersion": "4.0.2",
    "baseName": "jhipster",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "gradle",
    "enableSocialSignIn": false,
    "rememberMeKey": "be20d9ae8447ba5287bdc47245fcd9c8f727ebea",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "clientFramework": "angular2",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ]
  }
}
```

Yesterday I tried to get work yarn_install but with no success. If appears that it's not possible to get Jenkins pipeline work with yarn_install then in #5177 is changelog for npmInstall (unfortunately mix of yarn and npm) what I finally got work.